### ISSUE TITLE
question

### ISSUE THREAD

**Issue by iznil** [BIT-Jack/IETP#1]
Hello, may I ask if you can provide the data preprocessing file for this paper? I would like to know the code for your sampling section. This is my email address, 2076085831@qq.com, Thank you.

**Comment by BIT-Jack** [comment]
Hi~ I'm glad that you are interested in our work. We referred to the work "Convolutional Social Pooling for Vehicle Trajectory Prediction." CVPRW, 2018 (https://github.com/nachiket92/conv-social-pooling) for details of  data pre-processing, where the preprocess of NGSIM data is the same with ours.

**Comment by BIT-Jack** [comment]
I understand that, for the sampling section, you may want to know how to realize the random sampling. By turning the parameter of "shuffle" in PyTorch into "True", we can easily obtained the randomly sampled training sub-datasets.

**Comment by iznil** [comment]
Thank you for your response. I am trying to write random sampling code in Matlab based on the existing data preprocessing. TrainSet.mat has two variables: traj and tracks. When randomly sampling TrainSet, these two variables need to be matched, and I think it may be necessary to consider the continuity of the sampled data. Therefore, I think this part of the code is a bit difficult to write. If it is convenient, can you refer to your code in this part? esteem it  a favor.

**Comment by BIT-Jack** [comment]
Oh, I see. Instead of processing the raw data into .mat files, the more convenient way to process the random sampled dataset is using "shuffle=True" in the python file (based on the dataloader of PyTorch). 

You can run the python file "train_base_learner.py" for several times to obtained several base learners trained by randomly sampled data. Note the sentence `` trDataloader = DataLoader(trSet,batch_size=batch_size,shuffle=True,num_workers=8,collate_fn=trSet.collate_fn)``, which realize the random sampling in each training process.

In other words, we only need one group of ".mat" files which is preprocessed by the raw data. And we can obtain several sub-set during the process of using the DataLoader of pytorch.
